llarp/service/endpoint_types.hpp

```diff
@@ -54,6 +54,8 @@ namespace llarp
 
     using PathEnsureHook = std::function<void(Address, OutboundContext*)>;
 
+    using LNSNameCache = std::unordered_map<std::string, std::pair<Address, llarp_time_t>>;
```

Comment:
Should probably either make decayed hashset a templated type with an arbitrary container, or make a decayed map class.  Likely other things will come along (or are already implemented) that would benefit from such a class.

Reply:
refactored